### PR TITLE
Feature/navigation implementation

### PR DESCRIPTION
**Pull request by alexs-rufer**
### Add Navigation Support and Bottom Navigation Menu

**Description:**
This pull request introduces navigation support to the app, including the implementation of a bottom navigation menu. The changes made are inspired by the navigation patterns of the bootcamp. 
Key updates include:

- Updated `MainActivity` to initialize navigation with `NavHost`. To be updated when more screens are created.
- Implemented `BottomNavigationMenu` like in the bootcamp but using composable from Material3 (because they are not depreciated).
- Defined some navigation routes and actions in `NavigationActions.kt`. This is a start but can be modified in the future depending on the screens we will create.

- Updated `libs.versions.toml` with new navigation library versions.
- Added navigation dependencies in `build.gradle.kts`.

Next we should be able to define the navigation with different screens in MainActivity.kt like we did in the bootcamp.

Note: the first commit (765f0a0cf0e86a5e5d0e8fcf61e08865e463dad9) was an error, I did not want to commit it on this branch. This is why I corrected my mistake with the last commit (3aefcef9113a0e5e1f1b0305c40f3dfa4b79ed79) in which I deleted the two files related to the first commit.

### CODE REVIEW

**Review by LeaBlancher — COMMENTED** [review]
It looks good for me! Good job ;)

I made some suggestion on parts that concerns not yet implemented screens so it is not needed to update the code now, we could also update it when the screens will be implemented (the WELCOME and the PROFILE screeen).

I let @MPepito do a second review before merging.

File: app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt
Comment (alexs-rufer) [reply app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt:23]: This is a good point. I’ve added the two profile screens (create and edit), but I believe we’ll still need the Profile Screen as shown in the Figma. Feel free to tell me if you have any other suggestion!

File: app/src/main/java/com/android/sample/MainActivity.kt
Comment (alexs-rufer) [reply app/src/main/java/com/android/sample/MainActivity.kt:44]: Totally true, modified it!

**Review by aferrante1 — COMMENTED** [review]
The implementation seems solid. Just a note to keep in mind that the Top Destinations need to be different depending on whether we are on the Student or Tutor screen.

**Review by AlexisAllmnd — APPROVED** [review]
Good work, and everything seems correct. The implementation is close to the bootcamp version. However, the Routes and Screens will be modified later (for example, the welcome screen is the same as the auth screen).

**Review by AlexisAllmnd — APPROVED** [review]
Approving merge

**Review by AlexisAllmnd — APPROVED** [review]
Authorized merge after merge conflit with @alexs-rufer

**Pull request merged**
